### ISSUE TITLE
JSpecify: infer `@Nullable` type arguments for type variables from unmarked code

### ISSUE THREAD

**Issue by msridhar** [uber/NullAway#1178]
Example:
```java
@NullMarked
class SampleNullUnmarkedCall {

  @NullUnmarked
  static class Foo<T> {
    Foo(T t) {}

    static <U> Foo<U> id(U u) { return new Foo<>(u); }
  }

  static void test() {
    // should report no error
    Foo<@Nullable Object> x = Foo.id(null);
  }
}
```

Here, since `Foo` is `@NullUnmarked`, we can infer `@Nullable Object` as the type argument for the `Foo.id(null)` call, even though `U` (and `T`) do not have explicit `@Nullable` upper bounds.
